functions/xcluster/main.k

```diff
@@ -0,0 +1,184 @@
+import models.io.upbound.platform.azure.v1alpha1 as platformazurev1alpha1
+import models.io.upbound.platform.gitops.v1alpha1 as gitopsv1alpha1
+import models.io.upbound.platform.observe.v1alpha1 as observev1alpha1
+import models.io.upbound.platformref.azure.v1alpha1 as platformrefv1alpha1
+
+oxr = platformrefv1alpha1.XCluster{**option("params").oxr}
+_ocds = option("params").ocds # observed composed resources
+_dxr = option("params").dxr # desired composite resource
+dcds = option("params").dcds # desired composed resources
+
+_metadata = lambda name: str -> any {
+    { 
+        name = name
+        annotations = { "krm.kcl.dev/composition-resource-name" = name }
+    }
+}
+
+# Extract parameters from the observed composite resource (XR)
+_id = oxr.spec?.parameters?.id or ""
+_region = oxr.spec?.parameters?.region or ""
+_deletionPolicy = oxr.spec?.parameters?.deletionPolicy or "Delete"
+_providerConfigName = oxr.spec?.parameters?.providerConfigName or "default"
+_version = oxr.spec?.parameters?.version or "1.31"
```

Comment:
Defaults for versions seems not the right approach to me, why not use xrd defaults?